functionaltests/basic-testcase-8x_test.go

```diff
@@ -0,0 +1,130 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package functionaltests
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+	"go.uber.org/zap"
+	"go.uber.org/zap/zaptest"
+
+	"github.com/elastic/apm-server/functionaltests/internal/asserts"
+	"github.com/elastic/apm-server/functionaltests/internal/esclient"
+	"github.com/elastic/apm-server/functionaltests/internal/gen"
+	"github.com/elastic/apm-server/functionaltests/internal/terraform"
+	"github.com/elastic/go-elasticsearch/v8/typedapi/types"
+)
+
+// basicUpgradeTestCase8x is a basic functional test case that performs a
+// cluster upgrade between 2 specified versions.
```

Comment:
Q: what does 8x mean in the name? Is it a minor upgrade within 8, or just any upgrades (including major) that's from or to 8?

Reply:
I renamed it in [7239733](https://github.com/elastic/apm-server/pull/15836/commits/7239733c60844cd7dfda6631ac4c7988c14c8c18)

The original idea was to bind this test case to tests starting from any `8x` versions (as there are incompatibilities in how some area works). But in practice this would be confusing and not that effective.